### ISSUE TITLE
Video Desync on OBS 27.1.3

### ISSUE THREAD

**Issue by Dumedo** [Palakis/obs-ndi#691]
Same as @Shivaxi and @riggallegos wrote some days ago on closed issue https://github.com/Palakis/obs-ndi/issues/60
I have big video desync on OBS 27.1.3 on windows.
Previous OBS version was fine, yes a little delay in sync after a few hour but was still ok.
Now after a few minutes video is 1sec late and audio is ok.

Audio and video come from my gaming PC, on the bottom I have a dashboard coming from Simhub on gaming PC that go to obs streaming PC. When I change gear, audio is in sync with simhub dash, but on video gear change happen after 1 second.
Both pc are connected via ethernet to a switch. The switch is shutted down every day when I shut down my gaming PC.

This video is before the update at the beginning of a race, no desync
https://www.twitch.tv/videos/1157783609?collection=MSRe7QujnRaOWw&t=00h02m37s
And the same video later in the race as you can see in this one there is a slight video desync
https://www.twitch.tv/videos/1157783609?collection=MSRe7QujnRaOWw&t=00h40m49s

This is after the update at the beginning of my race https://www.twitch.tv/videos/1178418050?t=00h13m09s

Only way to reset video is to change from "Source timing" to "Network".. then it's fixed so you can go back to "Source timing".. but it is ok only for a few minutes, sorry this I haven't tested.

I double posted here because issue https://github.com/Palakis/obs-ndi/issues/60 is closed.

**Comment by Shivaxi** [comment]
Yeah thank you.  This is driving me crazy, I do the same thing to "reset" it, changing Sync from Source Timing to Network and then back again, or turning on and off Hardware Acceleration, whatever sort of "refreshes" it as it were.

Initially got the problem in 27.1.0.  I rolled back to my previous version 27.0.1, but that did not fix it for some reason.  I have tried leaving on Source Timing vs Networking, desync still happens after a while.  I have tried Latency Mode = Normal and Low, both still get the issue eventually, and actually Normal seems worse now, and I HAVE to use Low latency mode.  I have tried Allow Hardware Acceleration but this adds a ridiculous delay to the video for some reason, so I keep it off.

Essentially I have tried everything, even going to older OBS versions, with no resolve.  And it's strange the issue started with an OBS update, but then persisted even with a roll back.  Only thing I haven't tried is a full wipe of OBS and re-install, but I'd rather not do with all the custom profiles and plugins I have right now.  I stream every day so finding time to do such a thing is a bit of an endeavor.

Anyway, I somehow hope all this information helps in some way.  I see a lot of "fixes" for this issue if it's the other way around, where usually the audio is delayed BEHIND the video.  But this is not the case with this specific issue.  This is the VIDEO being delayed BEHIND the AUDIO instead.  I want that to be very clear, as some have linked me to fixes that were intended for the other way around.  And the desync will exponentially grow over time, getting worse and worse until it is "refreshed" as described above.

**Comment by Shivaxi** [comment]
> @Shivaxi just to double check, rembember cqp from gaming to stream pc. CBR from stream pc to twitch. it is not recommended to stream wit cqp. but you probably already know this! greetings!

So as far as I know, changing the quality settings on the main PC doesn't even do anything?  I can change the quality all the way down to fastest preset and like 1000 kbps, and it still looks perfectly fine over on the streaming PC.  I remember reading somewhere that NDI does not even look at your quality settings when outputting across your network, so whatever you set your Main PC really shouldn't matter, other than Encoder and Rescale Output, which are locked off if NDI is active.

I also hear a lot how separating video and audio NDI sources really helps but, the thing is, I've NEVER had to do this before.  So even if this did resolve the issue, it's not fixing the real problem, it's just going around it.  I haven't changed my OBS settings in almost 2 years, and only just now started having these issues after the latest OBS update as I stated before...it's bizarre.

Still haven't been able to fix this yet no matter what I do, but I'll keep testing.  Maybe I'll try the separate video/audio NDI sources for fun.  Though considering I have several scenes that I switch through on the fly on my Main PC while streaming, I still need to use the main NDI Output Settings to capture my video.

**Comment by Shivaxi** [comment]
> @Shivaxi so you didn't use any of @riggallegos advice?

No, I had tried his suggestions but none worked.  But also after further testing, delay still happens, its just not as noticeable as quick but it still can get very bad with long sessions, so turning off the streaming service requirements was only part of the issue.  I ended up reverting back to OBS 27.0.1 and that has entirely fixed my delay problem now.  So it seems OBS has just screwed up again with its update somewhere, and considering it only seems to be effecting 3rd party plugins like NDI I'm not so sure they're gonna fix it.  I'm just sticking with 27.0.1 for now.

**Comment by simplified-life** [comment]
> @simplified-life try this: [#691 (comment)](https://github.com/Palakis/obs-ndi/issues/691#issuecomment-945382359)
> 
You sir are a legend! This fixed my issue. Low latency mode on 27.1.3 has resulted in my video being predictably 10ms behind my audio and sources are in sync to the millisecond! So happy!

**Comment by Shivaxi** [comment]
> > @simplified-life try this: [#691 (comment)](https://github.com/Palakis/obs-ndi/issues/691#issuecomment-945382359)
> 
> You sir are a legend! This fixed my issue. Low latency mode on 27.1.3 has resulted in my video being predictably 10ms behind my audio and sources are in sync to the millisecond! So happy!

I was already using the low latency mode and this did not fix my issue for long streaming sessions.  I still had to revert to 27.0.1

**Comment by Shivaxi** [comment]
Okay, slight discovery.  on 27.0.1 and previous versions, the desync does still occur.  However, it automatically fixes itself, as periodically you can see a jump in the footage, where the video skips ahead to sync up properly with the audio.  This automatic jump ahead fix periodically does NOT happen in 27.1.1 and above, which I believe is why me and others really started to notice, so instead of it automatically fixing itself every hour or couple of hours, now it doesn't, and gets progressively worse and worse.

I hope somehow this information helps in some way.

**Comment by Shivaxi** [comment]
> Okay, slight discovery. on 27.0.1 and previous versions, the desync does still occur. However, it automatically fixes itself, as periodically you can see a jump in the footage, where the video skips ahead to sync up properly with the audio. This automatic jump ahead fix periodically does NOT happen in 27.1.1 and above, which I believe is why me and others really started to notice, so instead of it automatically fixing itself every hour or couple of hours, now it doesn't, and gets progressively worse and worse.
> 
> I hope somehow this information helps in some way.

To add a little, not sure if related, but when OBS on my main computer shuts off or crashes or I close it, it freeze frames over on the streaming OBS, in 27.0.1 and previous.  But with 27.1.1 and above, the screen just goes black.  Unsure if relevant or helpful.

**Comment by CicoZhang** [comment]
If NDI is used on the same PC with OBS, audio and video is in sync (audio is via audio interface and delay is set), but via Gigabit network, audio lags behind Video on the other computer (Mac, as the streaming machine), is there a fix to this?

**Comment by scottkorber** [comment]
I'm also experiencing this. I can run OBS on Computer A just playing a video, then pull the stream via NDI using OBS on Computer B and the audio/video sync will drift. I've also tried with Dicaffeine receiving the NDI stream with the same results. If I create a dedicated audio source from the videos audio on Computer A and pull that as a second NDI stream on Computer B and mute the audio being transmitted with the video, it will be in sync, and stay in sync.

**Comment by Dumedo** [comment]
@CicoZhang @scottkorber try "low latency Experimental mode" it worked for me.. also look at this post:
https://github.com/Palakis/obs-ndi/issues/691#issuecomment-945382359

**Comment by scottkorber** [comment]
> @CicoZhang @scottkorber try "low latency Experimental mode" it worked for me.. also look at this post: [#691 (comment)](https://github.com/Palakis/obs-ndi/issues/691#issuecomment-945382359)

I've already tried that. It doesn't change anything.

**Comment by Dumedo** [comment]
> 
> 
> > @CicoZhang @scottkorber try "low latency Experimental mode" it worked for me.. also look at this post: [#691 (comment)](https://github.com/Palakis/obs-ndi/issues/691#issuecomment-945382359)
> 
> I've already tried that. It doesn't change anything.

Try all the other advices, I already had advice 1 set on my streaming PC
